x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -349,11 +349,11 @@ export class KnowledgeBaseService {
   }
 
   private async getConnectorIndices(
-    client: ElasticsearchClient,
+    esClient: { asCurrentUser: ElasticsearchClient },
     uiSettingsClient: IUiSettingsClient
   ) {
     // improve performance by running this in parallel with the `uiSettingsClient` request
-    const responsePromise = client.transport.request({
+    const responsePromise = esClient.asCurrentUser.transport.request({
```

Comment:
does this mean the internal user does not have access to these?

Reply:
Yes, in this case we only pass `asCurrentUser` so it's not possible to call `esClient. asInternalUser` if that's what you mean

Reply:
I see. Yeah, good point. The end user may not have access to the `_connectors` endpoint and if the internal has we should probably make the request with that user. We'd possibly expose a list of index names to the user, even though they do not have access to them. Is that problematic?

Either way, I won't change that here but happy to create a follow-up PR

Reply:
> We'd possibly expose a list of index names to the user, even though they do not have access to them. Is that problematic?

Yeah that's fair. We should follow up with the Search team how exactly this works, maybe this endpoint doesn't require elevated privileges and this whole discussion is moot anyway.